parsing/parser.mly

```diff
@@ -1067,6 +1067,8 @@ The precedences must be listed from low to high.
 %inline mk_directive_arg(symb): symb
     { mk_directive_arg ~loc:$sloc $1 }
 
+%inline mark_payload_attrs_used(symb): symb
```

Comment:
This seems unnecessarily indirect. Just call `Builtin_attributes.mark_payload_attrs_used` in the body of `attr_payload`. No need for this parameterized parser non-terminal.

Actually, even better: could you use the new stateful approach to avoid adding to the table at all when in an attribute payload? I think you could, and that would save you the extra walk over the attribute payload in `mark_payload_attrs_used`.

Reply:
I didn't take the latter approach because I was trying to be a bit defensive: I don't fully understand the code menhir generates and was concerned about what might happen to the state in error cases.

Reply:
(But I guess if we use protect refs and put it just in the action, it's fine.  I'll make that change.)

Reply:
Sorry - I think that last comment is wrong.  So to explain in more detail why I didn't take the latter approach: It seems tricky - by the time we're in the action of the attribute nonterminal, it's too late to change the state.  So we need some more complicated set up with additional non terminals that guard the parsing of attributes.  This, combined with the fact that the action of those nonterminals would be to change state, seemed possible but hard to get right.

Reply:
(And as for the `mark_payload_attrs_used` bit - my impression was that this approach is cleaner and more idiomatic, but as you disagree I'll move it.)